src/storage-operators/src/persist_source.rs

```diff
@@ -492,7 +503,9 @@ impl PendingPart {
     fn part_mut(&mut self) -> (&mut FetchedPart<SourceData, (), Timestamp, Diff>, bool) {
         match self {
             PendingPart::Unparsed(x) => {
-                let error_free = error_free(x.stats()).unwrap_or(false);
+                // WIP hack
```

Comment:
Not quite sure yet what to do here. The optimization for ignore_data fetch works by taking advantage of the optimization for ignore_data decode and filling in placeholder data for key and val in a one-update inline part.

The issue is that the decode optimization needs to be able to prove that there are no errors, but inline parts don't have stats so it can't do that for the faked inline part. Some options:
- Use a key and val in the faked update that will actually decode. Then we don't need the decode optimization to fire. I don't want to put a `Default` bound on `Codec` (or all the methods in this codepath), so maybe stuff it in `ProjectionPushdown::IgnoreAllNonErr`?
- Somehow plumb around that the faked inline part is known to be error free
- Idk, something else